Issue/11899 new image selector

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #11899 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
Add all the logic to handle adding, replacing or removing and product image to be scanned and extract a prompt from it. 
Not that this PR only adds the logic to manage the image and update the UI accordingly, the scanning of the image will be tackled in a different PR. 

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
1. Trigger product creation with AI flow
2. Add an image by clicking on `Read text from product photo`
3. Check that the UI is updated with the image
4. Click on the three options available to manage the image: `view`, `replace`, `remove`. 

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/woocommerce/woocommerce-android/assets/2663464/e8897eb2-43e3-4dcf-b360-8e2188599723

JorgeMucientes committed:
Display media picker dialog

JorgeMucientes committed:
Display selected image

JorgeMucientes committed:
Add overflow menu to handle updating selected image

JorgeMucientes committed:
Fix image menu UI glitches

JorgeMucientes committed:
Handle image actions

JorgeMucientes committed:
Alternative way to view image full screen

JorgeMucientes committed:
Handle back navigation when image in full screen mode

dangermattic commented:
<!--
  0 Errors
  2 Warnings: This PR is larger than 300 lin...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>ImageAction</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Add missing copy

hafizrahman commented:
I don't think this should be a blocker for this PR, but I'd suggest the image preview functionality to either maintain the original image ratio even if it means zooming out the image, or stay as it is but allow users to tap and drag around to see the entire image.

In this PR's state, in my example image below, the image is shown cut-off and I am not able to see the rest of the details:

https://github.com/woocommerce/woocommerce-android/assets/266376/8c437489-73f5-4d73-b5ca-97fd96991ad2


I think it should not be a blocker because I don't think viewing the photo is a super critical flow. A merchant is supposed to pick a photo for the AI to read texts from, and being able to view a photo after adding it I suppose is only needed to double-check that that they've added the right image. But even if the image is wrong, they can always remove and retry, so the act of double-checking itself I suppose is not a critical feature.

hafizrahman commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/productinfo/AiProductPromptScreen.kt:
```diff
@@ -283,16 +339,107 @@ private fun ProductPromptTextField(
+                        text = stringResource(id = item.displayName),
+                        color = when (item) {
+                            ImageAction.Remove -> MaterialTheme.colors.error
+                            else -> Color.Unspecified
```
TIL about the use of `Color.Unspecified` 👍🏼

hafizrahman reviewed (approved):
This works nicely in my tests, and the code looks solid, too. I left some non-blocking comments, feel free to merge after reading them.

hafizrahman commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ai/productinfo/AiProductPromptFragment.kt:
```diff
@@ -42,7 +53,47 @@ class AiProductPromptFragment : BaseFragment() {
             when (event) {
                 Exit -> findNavController().navigateUp()
+
+                is ViewImage -> showImageDetail(event.mediaUri)
```
❓  This does not seem to be used anymore / is replaced by the Compose `FullScreenImage` component?

JorgeMucientes committed:
Remove unused code

JorgeMucientes committed:
Avoid cropping image and making it fit the screen in full screen mode

JorgeMucientes commented:
Great feedback @hafizrahmanm thanks! I've addressed both of the shared points. 

- Removed unused code from prior approach for opening the image on full screen
- Avoid cropping the image when opening it in full screen while always filling the available space but without altering the aspect ratio. 

https://github.com/woocommerce/woocommerce-android/assets/2663464/3e82f5e1-4618-4f82-bcde-46af8b213312
